src/parser.ts

```diff
@@ -40,71 +190,48 @@ export interface CodeSample {
 
 export type MimeType = 'application/json' | 'text/plain; charset=utf-8' | 'application/x-binary'
 
-export interface Schema {
-  description: string;
-  type?: XtpSchemaType;
-  enum?: string[];
-  properties?: { [name: string]: Property };
-  additionalProperties?: Property;
+export interface Schema extends XtpTyped {
+  description?: string;
   required?: string[];
+  properties?: { [name: string]: Property };
 }
 
-export type XtpSchemaType = 'object' | 'enum' | 'map'
+// TODO this figure out how to split up type again?
+//export type XtpSchemaType = 'object' | 'enum' | 'map'
```

Comment:
we don't really need this type anymore. we can now put any type into a schema. though maybe we should hold off on relaxing the constraints in the json schema until we can test. In theory we should be able to support enums and maps as properties too (inline definition) but we'll want to wait to do that and will need to add a requirement that they have name.